src/main/java/com/iabtcf/v2/BitVectorTCModelV2.java

```diff
@@ -19,296 +19,352 @@
-        this.isPurposeOneTreatment = bv.readBits1(PURPOSE_ONE_TREATMENT_OFFSET);
-        this.publisherCountryCode = readStr2(bv, PUBLISHER_CC_OFFSET);
-
-        int currentPointer = PUBLISHER_CC_OFFSET + (CHAR.length() * 2); // publisher cc offset
-        this.vendorConsents = new TreeSet<>();
-        currentPointer = this.fetchSet(this.vendorConsents, currentPointer, bv);
-
-        this.vendorLegitimateInterests = new TreeSet<>();
-        currentPointer = this.fetchSet(this.vendorLegitimateInterests, currentPointer, bv);
-
-        this.publisherRestrictions = new ArrayList<>();
-        this.fillPublisherRestrictions(publisherRestrictions, currentPointer, bv);
-
-        this.disclosedVendors = new TreeSet<>();
-        this.allowedVendors = new TreeSet<>();
-        this.publisherPurposesConsent = new TreeSet<>();
-        this.publisherPurposesLITransparency = new TreeSet<>();
-        this.customPurposesConsent = new TreeSet<>();
-        this.customPurposesLITransparency = new TreeSet<>();
-    }
-
-    private BitVectorTCModelV2(ByteBitVector coreVector, ByteBitVector... theRest) {
-        this(coreVector);
-        for (ByteBitVector bitVector : theRest) {
-            fillRemainingVector(bitVector);
-        }
-    }
-
-    public static BitVectorTCModelV2 fromBitVector(
-            ByteBitVector coreBitVector, ByteBitVector... remainingVectors) {
+    private int version;
+    private Instant consentRecordCreated;
+    private Instant consentRecordLastUpdated;
+    private int consentManagerProviderId;
+    private int consentManagerProviderVersion;
+    private int consentScreen;
+    private String consentLanguage;
+    private int vendorListVersion;
+    private int policyVersion;
+    private boolean isServiceSpecific;
+    private boolean useNonStandardStacks;
+    private Set<Integer> specialFeaturesOptInts;
+    private Set<Integer> purposesConsent;
+    private Set<Integer> purposesLITransparency;
+    private boolean isPurposeOneTreatment;
+    private String publisherCountryCode;
+    private Set<Integer> vendorConsents;
+    private Set<Integer> vendorLegitimateInterests;
+    private List<PublisherRestriction> publisherRestrictions;
+    private Set<Integer> disclosedVendors;
+    private Set<Integer> allowedVendors;
+
+    private Set<Integer> publisherPurposesConsent;
+    private Set<Integer> publisherPurposesLITransparency;
+    private Set<Integer> customPurposesConsent;
+    private Set<Integer> customPurposesLITransparency;
+
+    private final EnumSet<FieldDefs> e = EnumSet.noneOf(FieldDefs.class);
+    private final ByteBitVector bbv;
+    private Collection<ByteBitVector> remainingVectors;
+
+    private BitVectorTCModelV2(ByteBitVector bbv) {
+        this(bbv, new ByteBitVector[] {});
+    }
+
+    private BitVectorTCModelV2(ByteBitVector bbv, ByteBitVector... theRest) {
+        this.bbv = bbv;
+        this.remainingVectors = Arrays.asList(theRest);
```

Comment:
Should just create this list in the TCModelDecoder instead of passing in an array and re-wrapping in a list.

Reply:
the whole mechanism with how we handle remainingVectors should be re-thought. I think it's out-side the scope of this commit however. I'll create an issue for this.